### DISCUSSION TITLE
ru/modes/track/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#10081] [Docs]
# ru/modes/track/

Узнай, как использовать Ultralytics YOLO  для отслеживания объектов в видеопотоке. Руководства по использованию различных трекеров и настройке конфигураций трекеров.

https://docs.ultralytics.com/ru/modes/track/

**Comment by mihkriss** [comment]
Hi, I’m doing a project to identify fights using key points and tracking, could you help me explain how this can be implemented. In my project, I implemented the extraction of skeletal points of fighting people, is it possible to track these skeletons?
Or extract tracking information from successive frames in a fight video?

**Reply by mihkriss** [reply]
Let me add how to get tracking values between frames or skeleton points?
to see their meanings

**Reply by glenn-jocher** [reply]
Hey there! 😊

For tracking skeletal points or extracting tracking info between frames in fight videos, you can indeed leverage the tracking capabilities of YOLOv8 with modifications to focus on keypoints. Here's a simplified approach:

1. **Keypoint Detection:** First, ensure your YOLO model is trained to recognize and output keypoints of interest (like skeletal points in a fight scenario). YOLOv8's Pose estimation models, such as `yolov8n-pose.pt`, can be a good starting point if adapted for your specific use case.

2. **Tracking:** Utilize the tracking feature to maintain identity across frames. YOLOv8 supports multiple trackers like BoT-SORT or ByteTrack, which you can activate with something like:
   ```python
   from ultralytics import YOLO

   model = YOLO('path/to/your/custom_pose_model.pt')  # Load your custom model
   results = model.track(source="path/to/fight_video.mp4", tracker="bytetrack.yaml")
   ```

3. **Analyzing Movement:** With the tracking data, you can analyze the keypoints' movement over time to identify patterns indicative of a fight. This might involve calculating the velocity or acceleration of certain keypoints, or the relative distances between keypoints over successive frames.

Remember, the model's performance, in this case, heavily relies on the quality of keypoints it's trained to recognize and the robustness of the tracking algorithm in maintaining identities across frames. Additionally, because each fight scenario can be significantly different, having a diverse and representative training dataset is crucial for accuracy.

Hope this helps! Keep up the great work with your project. 🌟

**Reply by mihkriss** [reply]
Where is tracking information stored, how can this data be retrieved?

**Reply by glenn-jocher** [reply]
Hey there! 🌟

Tracking information like IDs and positions are accessible through the `results` object returned after running a track operation. For skeletal point tracking or any keypoints, you can find this data within the `results.keypoints` attribute after tracking.

Here's a quick peek at how you might retrieve and use this information:

```python
from ultralytics import YOLO

# Load your tracking model
model = YOLO('path/to/your/model.pt')

# Perform tracking
results = model.track(source='path/to/your/video.mp4')

# Iterate through results to access tracking information
for result in results:
    keypoints = result.keypoints  # accessing keypoints for each tracked object
    for individual in keypoints:
        print(individual.xy)  # x, y coordinates
```

This will give you the tracked keypoints' positions. Remember, your model should be trained to detect and track such keypoints for accurate results.

Hope this points you in the right direction! 😉 Happy tracking!
